src/TypeParser/JMSTypeParser.php

```diff
@@ -17,6 +17,7 @@
 final class JMSTypeParser
 {
     private const TYPE_ARRAY = 'array';
+    private const TYPE_ARRAY_COLLECTION = 'ArrayCollection';
```

Comment:
should we detect any `ArrayAccess` as array? 

the problem however is that then we would need a hard dependency on doctrine collections to be sure we have something we can instantiate - we can't instantiate generically an unknown type of array access as we don't know its constructor signature.

did you look into this? or should we just leave it at the doctrine class for now? it should be simple to expand later on if somebody has the need.

Reply:
thinking of the serializer as well: lets drop this idea - doctrine Collection is good enough. generic array access does not tell how to serialize it either, so its not really feasible.